Clippy errors when running CI

There are new clippy errors when running the workflows:


```rust
run cargo clippy --tests --benches --examples --workspace --all-targets --all-features
  cargo clippy --tests --benches --examples --workspace --all-targets --all-features
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    CARGO_HOME: /home/runner/.cargo
    CARGO_INCREMENTAL: 0
    CACHE_ON_FAILURE: false
   Compiling torrust-index v3.0.0-develop (/home/runner/work/torrust-index/torrust-index)
    Checking torrust-index-located-error v3.0.0-develop (/home/runner/work/torrust-index/torrust-index/packages/located-error)
error: unnecessary semicolon
   --> src/console/commands/seeder/app.rs:199:10
    |
199 |         };
    |          ^ help: remove
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon
    = note: `-D clippy::unnecessary-semicolon` implied by `-D clippy::pedantic`
    = help: to override `-D clippy::pedantic` add `#[allow(clippy::unnecessary_semicolon)]`

error: unnecessary semicolon
  --> src/mailer.rs:28:10
   |
28 |         };
   |          ^ help: remove
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon

error: `format!(..)` appended to existing `String`
   --> src/services/torrent.rs:553:13
    |
553 |             magnet.push_str(&format!("&tr={}", urlencoding::encode(tracker)));
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: consider using `write!` to avoid the extra allocation
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#format_push_string
    = note: `-D clippy::format-push-string` implied by `-D clippy::pedantic`
    = help: to override `-D clippy::pedantic` add `#[allow(clippy::format_push_string)]`

error: unnecessary semicolon
   --> src/web/api/server/v1/contexts/torrent/handlers.rs:339:18
    |
339 |                 };
    |                  ^ help: remove
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon

error: unnecessary semicolon
   --> src/web/api/server/v1/contexts/torrent/handlers.rs:346:18
    |
346 |                 };
    |                  ^ help: remove
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon

error: unnecessary semicolon
   --> src/web/api/server/v1/contexts/torrent/handlers.rs:353:18
    |
353 |                 };
    |                  ^ help: remove
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon

error: unnecessary semicolon
   --> src/web/api/server/v1/contexts/torrent/handlers.rs:360:18
    |
360 |                 };
    |                  ^ help: remove
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon

error: unnecessary semicolon
   --> src/web/api/server/mod.rs:110:6
    |
110 |     };
    |      ^ help: remove
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon

error: could not compile `torrust-index` (lib) due to 8 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `torrust-index` (lib test) due to 8 previous errors
Error: Process completed with exit code 101.
```

--- a/src/bin/create_test_torrent.rs
+++ b/src/bin/create_test_torrent.rs
@@ -60,15 +60,13 @@ fn main() {
             // Attempt to create and write to the file
             let mut file = match File::create(&file_path) {
                 Ok(file) => file,
-                Err(e) => panic!("Failed to create file {file_path:?}: {e}"),
+                Err(e) => panic!("Failed to create file {}: {e}", file_path.display()),
             };
 
             if let Err(e) = file.write_all(&bytes) {
-                panic!("Failed to write to file {file_path:?}: {e}");
+                panic!("Failed to write to file {}: {e}", file_path.display());
             }
-
-            println!("File successfully written to {file_path:?}");
         }
         Err(e) => panic!("Error encoding torrent: {e}"),
-    };
+    }
 }

--- a/src/console/commands/seeder/app.rs
+++ b/src/console/commands/seeder/app.rs
@@ -196,7 +196,7 @@ pub async fn run() -> anyhow::Result<()> {
                 info!(target:"seeder", "Uploaded torrent: {}", json.yellow());
             }
             Err(err) => print!("Error uploading torrent {err:?}"),
-        };
+        }
 
         if i != args.number_of_torrents {
             sleep(Duration::from_secs(args.interval));

--- a/src/mailer.rs
+++ b/src/mailer.rs
@@ -25,7 +25,7 @@ lazy_static! {
                 println!("Parsing error(s): {e}");
                 ::std::process::exit(1);
             }
-        };
+        }
 
         tera.autoescape_on(vec![".html", ".sql"]);
         tera.register_filter("do_nothing", do_nothing_filter);

--- a/src/services/torrent.rs
+++ b/src/services/torrent.rs
@@ -1,4 +1,5 @@
 //! Torrent service.
+use std::fmt::Write as _;
 use std::sync::Arc;
 
 use bittorrent_primitives::info_hash::InfoHash;
@@ -550,7 +551,7 @@ impl Index {
 
         // Add trackers from torrent file to magnet link
         for tracker in &torrent_response.trackers {
-            magnet.push_str(&format!("&tr={}", urlencoding::encode(tracker)));
+            let _ = write!(&mut magnet, "&tr={}", urlencoding::encode(tracker));
         }
 
         torrent_response.magnet_link = magnet;

--- a/src/services/user.rs
+++ b/src/services/user.rs
@@ -206,7 +206,7 @@ impl RegistrationService {
 
         if self.user_profile_repository.verify_email(&user_id).await.is_err() {
             return Err(ServiceError::DatabaseError);
-        };
+        }
 
         Ok(true)
     }

--- a/src/web/api/server/mod.rs
+++ b/src/web/api/server/mod.rs
@@ -107,7 +107,7 @@ async fn start_server(
             .serve(router.into_make_service_with_connect_info::<std::net::SocketAddr>())
             .await
             .expect("API server should be running"),
-    };
+    }
 }
 
 #[derive(Error, Debug)]

--- a/src/web/api/server/v1/contexts/torrent/handlers.rs
+++ b/src/web/api/server/v1/contexts/torrent/handlers.rs
@@ -336,28 +336,28 @@ async fn build_add_torrent_request_from_payload(mut payload: Multipart) -> Resul
                 let data = field.bytes().await.unwrap();
                 if data.is_empty() {
                     continue;
-                };
+                }
                 title = String::from_utf8(data.to_vec()).map_err(|_| errors::Request::TitleIsNotValidUtf8)?;
             }
             "description" => {
                 let data = field.bytes().await.unwrap();
                 if data.is_empty() {
                     continue;
-                };
+                }
                 description = String::from_utf8(data.to_vec()).map_err(|_| errors::Request::DescriptionIsNotValidUtf8)?;
             }
             "category" => {
                 let data = field.bytes().await.unwrap();
                 if data.is_empty() {
                     continue;
-                };
+                }
                 category = String::from_utf8(data.to_vec()).map_err(|_| errors::Request::CategoryIsNotValidUtf8)?;
             }
             "tags" => {
                 let data = field.bytes().await.unwrap();
                 if data.is_empty() {
                     continue;
-                };
+                }
                 let string_data = String::from_utf8(data.to_vec()).map_err(|_| errors::Request::TagsArrayIsNotValidUtf8)?;
                 tags = serde_json::from_str(&string_data).map_err(|_| errors::Request::TagsArrayIsNotValidJson)?;
             }

--- a/tests/e2e/environment.rs
+++ b/tests/e2e/environment.rs
@@ -84,7 +84,7 @@ impl TestEnv {
     pub fn provides_a_private_tracker(&self) -> bool {
         if !self.is_shared() {
             return false;
-        };
+        }
 
         match self.server_settings() {
             Some(settings) => settings.tracker.private,
@@ -148,11 +148,7 @@ impl TestEnv {
             State::RunningShared => {
                 let connect_url_env_var = ENV_VAR_DB_CONNECT_URL;
 
-                if let Ok(connect_url) = env::var(connect_url_env_var) {
-                    Some(connect_url)
-                } else {
-                    None
-                }
+                env::var(connect_url_env_var).ok()
             }
             State::RunningIsolated => internal_connect_url,
             State::Stopped => None,

